tools/codegen/api/types.py

```diff
@@ -72,7 +72,69 @@ def cpp_type(self, *, strip_ref: bool = False) -> str:
     def name(self) -> ArgName:
         return self.elem.name
 
-CType = Union[BaseCType, OptionalCType, ConstRefCType, MutRefCType]
+@dataclass(frozen=True)
+class ListCType:
+    elem: 'CType'
+
+    def cpp_type(self, *, strip_ref: bool = False) -> str:
+        # Do not pass `strip_ref` recursively.
```

Comment:
man `strip_ref` was a mistake haha. Since most of the time it doesn't go recursively maybe we can just make a small recursive pattern matching function that handles this.